classes/base_decorator.rb

```diff
@@ -0,0 +1,13 @@
+require_relative 'nameable'
+
+# class Decorator
+class Decorator < Nameable
```

Comment:
- [ ] Getters and Setters for `nameable` should be present to allow us to access and update the `:nameable` from outside the scope of the class definition. Kindly add this 👇🏻 

`attr_accessor :nameable`